src/main/java/com/connexta/search/index/IndexServiceImpl.java

```diff
@@ -6,165 +6,24 @@
-    } catch (final Exception e) {
-      throw new IndexException(INTERNAL_SERVER_ERROR, "Unable to save index", e);
-    }
-  }
-
-  @Override
-  public Set<String> query(String cql) throws IndexException {
-    return this.query(cql, new FilterToSolr(null));
-  }
-
-  @VisibleForTesting
-  Set<String> query(String cql, FilterToSolr filterToSolr) {
-    Filter filter;
-    try {
-      filter = ECQL.toFilter(cql);
-    } catch (CQLException e) {
-      try {
-        // The ECQL class does not support an equals on id, ie `id=foo`, but CQL does.
-        filter = CQL.toFilter(cql);
-      } catch (CQLException cqle) {
-        log.debug("Invalid CQL received: {}", cql, cqle);
-        throw new IndexException(BAD_REQUEST, "Invalid CQL received");
-      }
-    }
-
-    final Set<String> unsupportedAttributes =
-        Filters.propertyNames(filter).stream()
-            .map(PropertyName::getPropertyName)
-            .filter(attribute -> !QUERY_TERMS.contains(attribute))
-            .collect(Collectors.toSet());
-    if (!unsupportedAttributes.isEmpty()) {
-      throw new IndexException(BAD_REQUEST, "Received invalid attributes to index on");
-    }
-
-    String solrQueryStr;
-    try {
-      solrQueryStr = filterToSolr.encodeToString(filter);
-    } catch (Exception e) {
-      log.debug("Unable to transform CQL {} to valid Solr query", cql, e);
-      throw new IndexException(INTERNAL_SERVER_ERROR, "Error processing CQL");
-    }
-
-    SolrQuery solrQuery = new SolrQuery(solrQueryStr);
-    QueryResponse response;
-    try {
-      response = solrClient.query(SolrConfiguration.SOLR_COLLECTION, solrQuery);
-    } catch (SolrServerException | IOException e) {
-      log.debug("Failed to query solr", e);
-      throw new IndexException(INTERNAL_SERVER_ERROR, "Error querying index");
-    }
-
-    Set<String> ids = new HashSet<>();
-    for (SolrDocument doc : response.getResults()) {
-      Object obj = doc.get(SolrConfiguration.ID_ATTRIBUTE);
-      if (obj instanceof String) {
-        ids.add((String) doc.get(SolrConfiguration.ID_ATTRIBUTE));
-      } else {
-        // shouldn't hit this since the schema enforces the ID is a string
-        log.debug("Skipping invalid solr result {}", doc);
-      }
-    }
-
-    return ids;
-  }
-
-  private static JsonNode parseJson(InputStream stream) throws IOException {
-    final ObjectMapper objectMapper;
-    objectMapper = new ObjectMapper();
-    return objectMapper.readTree(stream);
-  }
-
-  private static String getElement(JsonNode json, String fieldName) throws IndexException {
-    if (json.get(fieldName) == null) {
-      throw new IndexException(INTERNAL_SERVER_ERROR, "JSON is malformed");
-    }
-    return json.get(fieldName).asText();
+      @Pattern(regexp = "^[0-9a-zA-Z]+$") @Size(min = 32, max = 32) final String datasetId,
+      @NotBlank final String contentType,
+      @NotNull final InputStream inputStream) {
+    searchManager.index(datasetId, contentType, inputStream);
```

Comment:
I can't see a purpose for this IndexService, it's simply become a wrapper and level of indirection around the SearchManager.

Reply:
See `IndexServiceImpl` in https://github.com/connexta/ion-search/pull/66. It is no longer just a wrapper.